src/BlazorWebView/src/Microsoft.AspNetCore.Components.WebView.Maui/Windows/WebView2WebViewManager.cs

```diff
@@ -0,0 +1,153 @@
+
+					// NOTE: This is a workaround for an issue with earlier builds of Blazor WebView/Desktop
+					// where the wrong MIME type is used, so we overwrite it.
+					if (allowFallbackOnHostPage)
+					{
+						headers = headers.Replace("application/octet-stream", "text/html");
+					}
+					else
+					{
+						if (eventArgs.Request.Uri.Contains("css", StringComparison.Ordinal))
+						{
+							headers = headers.Replace("application/octet-stream", "text/css");
+						}
+					}
+
+					eventArgs.Response = environment.CreateWebResourceResponse(ms, statusCode, statusMessage, headers);
+				}
+			};
+
+			// The code inside blazor.webview.js is meant to be agnostic to specific webview technologies,
+			// so the following is an adaptor from blazor.webview.js conventions to WebView2 APIs
+			await _webview.CoreWebView2!.AddScriptToExecuteOnDocumentCreatedAsync(@"
+                window.external = {
+                    sendMessage: message => {
+                        window.chrome.webview.postMessage(message);
+                    },
+                    receiveMessage: callback => {
+                        window.chrome.webview.addEventListener('message', e => callback(e.data));
+                    }
+                };
+            ");
+
+			_webview.CoreWebView2.WebMessageReceived += (sender, eventArgs)
+				=> MessageReceived(new Uri(eventArgs.Source), eventArgs.TryGetWebMessageAsString());
+
+			_webview.CoreWebView2.DOMContentLoaded += async (_, __) =>
+			{
+				await _webview.CoreWebView2!.ExecuteScriptAsync(@"
+					Blazor.start();
+					");
+			};
+		}
+
+		//private static string GetHeaderString(IDictionary<string, string> headers) =>
+		//	string.Join(Environment.NewLine, headers.Select(kvp => $"{kvp.Key}: {kvp.Value}"));
+
+		private void ApplyDefaultWebViewSettings()
+		{
+			// Desktop applications typically don't want the default web browser context menu
+			_webview.CoreWebView2.Settings.AreDefaultContextMenusEnabled = false;
+
+			// Desktop applications almost never want to show a URL preview when hovering over a link
+			_webview.CoreWebView2.Settings.IsStatusBarEnabled = false;
+
+			// TODO: This event doesn't exist in WinUI
+
+			// Desktop applications don't normally want to enable things like "alt-left to go back"
+			// or "ctrl+f to find". Developers should explicitly opt into allowing these.
+			// Issues #30511 and #30624 track making an option to control this.
+			//_webview.AcceleratorKeyPressed += (sender, eventArgs) =>
+			//{
+			//	if (eventArgs.VirtualKey != 0x49) // Allow ctrl+shift+i to open dev tools, at least for now
+			//	{
+			//		// Note: due to what seems like a bug (https://github.com/MicrosoftEdge/WebView2Feedback/issues/549),
+			//		// setting eventArgs.Handled doesn't actually have any effect in WPF, even though it works fine in
+			//		// WinForms. Leaving the code here because it's supposedly fixed in a newer version.
+			//		eventArgs.Handled = true;
+			//	}
+			//};
+		}
+	}
```

Comment:
I'm assuming it will be possible to not duplicate all this code, will it?

Reply:
I think I mentioned this briefly in the call, but it's worth putting everything here:

The assembly identity of all the WebView2 types is different between WinUI and WPF/WinForms. As such, there is no direct way to re-use the existing WebView2 helper types from the ASP.NET Core repo. One way to re-use the types is to... provide yet another abstraction, and implement one for WinUI, and one for WPF/WinForms.

I plan to file an issue for the WebView2 folks because it seems odd to me that the "same" types have different identity depending on the platform, meaning no one using/extending WebView2 can share code at all.